### PR TITLE
Remove the chevron icons from the “Finish setup” task list

### LINKED ISSUE #4896: Remove chevrons from Task List items
As per [this update](https://github.com/woocommerce/woocommerce-admin/pull/4790) of the Store Management card, let's remove the chevrons from the Task List links as well.

remove-chevrons

### PR DESCRIPTION
**Pull request by adrianduffell**
Fixes #4896

This aligns with the update of the Store Management card in #4790 by removing the chevron icon from the item rows in this task list.

Now, only a "Dismiss" button renders in the right-side area of a row. This occurs if the task is optional and incomplete. Due to this a small refactor is made to remove the conditional operator when assigning `task.after`.

A test is added to assert the logic for rendering the "Dismiss" button.

### Screenshots

Screen Shot 2020-09-09 at 1 48 10 pm

### Detailed test instructions:

- Start on a site that hasn’t finished the setup tasks, and has the "Purchase & install extensions" task still to do.
- Confirm that chevron icons have been removed from the right-hand side of each incomplete task.
- Confirm that a "Dismiss" button is still present for the  "Purchase & install extensions" task.

### Changelog Note:

Tweak: The "Finish setup" task list no longer displays a chevron icon for incompleted tasks.

### CODE REVIEW

File: client/task-list/test/index.js [hunk 1]
```diff
@@ -0,0 +1,72 @@
...
+				completed: true,
+				visible: true,
+				time: '1 minute',
+				isDismissable: true,
+			},
+		] ),
+	};
+} );
+
+describe( 'TaskDashboard', () => {
```
Comment (samueljseay) [inline client/task-list/test/index.js:50]: Great to see some testing 🎉

File: client/task-list/test/index.js [hunk 2]
```diff
@@ -0,0 +1,72 @@
...
+	afterEach( () => jest.clearAllMocks() );
+
+	it( 'renders a dismiss button for tasks that are optional and incomplete', async () => {
+		apiFetch.mockResolvedValue( {} );
+
+		render( <TaskDashboard query={ {} } /> );
+
+		// Wait for the task list to render.
+		expect( await screen.findByText( 'Finish setup' ) ).toBeDefined();
```
Comment (samueljseay) [inline client/task-list/test/index.js:59]: Not requesting a change but just an FYI that you mostly never need to query the whole screen for component tests. Rather you can capture the container when calling render and then query that.

e.g.

```
const {container} = render(<TaskDashboard query={{}} />);

expect( container.findByText('Finish setup')).toBeDefined();
```

File: client/task-list/index.js [hunk 3]
```diff
@@ -298,9 +298,10 @@ class TaskDashboard extends Component {
 				</Text>
 			);
 
-			if ( ! task.completed ) {
-				task.after = task.isDismissable ? (
+			if ( ! task.completed && task.isDismissable ) {
```
Comment (samueljseay) [inline client/task-list/index.js:301]: nice change 😄

File: client/task-list/test/index.js (hunk 2 above)
Comment (adrianduffell) [reply client/task-list/test/index.js:59]: Thanks! I'll make this change.

**Review by samueljseay — APPROVED** [review]
Pre-approving this. I've tested the code locally and it works well.

File: client/task-list/test/index.js (hunk 2 above)
Comment (adrianduffell) [reply client/task-list/test/index.js]: I am querying the container now in 985f8d3e39b9012db51fd65de74a05db76878f5b 😄

**Pull request merged**